include/xgboost/data.h

```diff
@@ -50,6 +50,8 @@ struct MetaInfo {
   std::vector<bst_uint> group_ptr;
   /*! \brief weights of each instance, optional */
   std::vector<bst_float> weights;
+  /*! \brief session-id of each instance, optional */
+  std::vector<bst_float> qids;
```

Comment:
Are you sure these should be floats?

Reply:
Thinks, I will change it to bst_uint.  Type bst_float also works, but bst_uint should be enough.